core/client/app/routes/posts.js

```diff
@@ -69,6 +69,22 @@ export default AuthenticatedRoute.extend(ShortcutsRoute, PaginationRouteMixin, {
     },
 
     actions: {
+        willTransition(transition) {
+            let transitionDueToDeletion = this.get('controller.currentPost.isDeleted')
+                && transition.targetName === 'posts.index';
+            let posts, nextPost, deletedPostIndex;
+
+            if (transitionDueToDeletion) {
+                posts = this.get('controller.sortedPosts');
+                deletedPostIndex = this.get('controller.currentPostIndex');
+                nextPost = posts.objectAt(deletedPostIndex);
+                if (nextPost) {
+                    return this.transitionTo('posts.post', nextPost);
```

Comment:
You might want to abort the existing transition here and then try redirecting to the `nextPost.id` so that the relevant model hooks in the post controller get called - not tested it but just a hunch that it might fix the `active` link issue.